sqswatcher/sqswatcher.py

```diff
@@ -148,29 +148,33 @@ def pollQueue(scheduler, q, t, proxy_config):
                             instance = next(iter(instances or []), None)
 
                             if not instance:
-                                log.warning("Unable to find running instance %s." % instanceId)
+                                log.error("Unable to find running instance %s." % instanceId)
                             else:
                                 hostname = instance.private_dns_name.split('.')[:1][0]
-                                log.info("Adding Hostname: %s" % hostname)
-                                s.addHost(hostname=hostname, cluster_user=cluster_user, slots=slots)
-
-                                t.put_item(Item={
-                                    'instanceId': instanceId,
-                                    'hostname': hostname
-                                })
-
-                            message.delete()
+                                if hostname:
+                                    log.info("Adding Hostname: %s" % hostname)
+                                    s.addHost(hostname=hostname, cluster_user=cluster_user, slots=slots)
+
+                                    t.put_item(Item={
+                                        'instanceId': instanceId,
+                                        'hostname': hostname
+                                    })
+                                else:
+                                    log.error("Unable to get the hostname for the instance %s." % instanceId)
                             break
                         except ClientError as e:
                             if e.response.get('Error').get('Code') == 'RequestLimitExceeded':
+                                log.debug("Request limit exceeded, waiting other %s seconds..." % wait)
                                 time.sleep(wait)
                                 retry += 1
-                                wait = (wait*2+retry)
+                                wait = wait * 2
                             else:
-                                raise e
+                                log.critical(e.response.get("Error").get("Message"))
+                                break
                         except:
                             log.critical("Unexpected error: %s" % sys.exc_info()[0])
-                            raise
+                            break
+                    message.delete()
```

Comment:
This effectively adds a re-try for any message that fails. Can you also add the retry logic to `autoscaling:EC2_INSTANCE_TERMINATE`? Say for example the dynamo call fails, we should re-try.